core/mutator/mutator.go

```diff
@@ -50,10 +51,13 @@ var (
 // VerifyMutationFn verifies that a mutation is internally consistent.
 type VerifyMutationFn func(mutation *pb.SignedEntry) error
 
-// ReduceMutationFn takes the existing mapleaf and a new mutation and returns the new value for that map leaf.
-// ReduceMutationFn verifies that this is a valid mutation for this item.
-// ReduceMutationFn must be idempotent.
-type ReduceMutationFn func(existingMapLeafValue, mutation *pb.SignedEntry) (*pb.SignedEntry, error)
+// ReduceMutationFn takes all the mutations for an index and an auxiliary input
+// of existing mapleaf(s) returns a new map leaf.  ReduceMutationFn must be
+// idempotent, commutative, and associative. i.e. must produce the same output
```

Comment:
It's no longer clear to me this morning how "associative" can be sensibly interpreted for a function that returns a different type than it consumes.

Reply:
I'll save this for the next PR :-)